### PR TITLE
Add a NetCDF extern interface module

### PR DESCRIPTION
**Pull request by daviditen**
The NetCDF.C_NetCDF module was generated by running c2chapel on NetCDF's main
include file, netcdf.h. I then created the modules around it and deleted
some leftover from the "fake headers".  I also manually added extern declarations for
all #define'd literals from netcdf.h.

I ported all of the examples from the examples section of the netCDF website to
ensure that the interface was working as expected.

### CODE REVIEW

File: test/library/packages/NetCDF/pres_temp_4D_rd.chpl [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+/* These are used to calculate the values we expect to find. */
+const samplePressure = 900.0,
+      sampleTemp = 9.0,
+      startLat = 25.0,
+      startLon = -125.0;
+
+proc cdfError(e) {
+  if e != NC_NOERR {
+    writeln("Error: ", nc_strerror(e): string);
```
[the remark this answers is not in the document]
Comment (daviditen) [reply test/library/packages/NetCDF/pres_temp_4D_rd.chpl]: IMO print an error and exit is better from this user-level, non-library code that's effectively in `main`.

File: test/library/packages/NetCDF/pres_temp_4D_rd.chpl [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   cdfError(nc_get_var_float(ncid, lat_varid, lats[0]));
+   cdfError(nc_get_var_float(ncid, lon_varid, lons[0]));
+
+   /* Check the coordinate variable data. */
+   for lat in 0..#nlat do
+      if (lats[lat] != startLat + 5.0*lat) then
+	 return 2;
+   for lon in 0..#nlon do
+      if (lons[lon] != startLon + 5.0*lon) then
+	 return 2;
```
[the remark this answers is not in the document]
Comment (daviditen) [reply test/library/packages/NetCDF/pres_temp_4D_rd.chpl]: Printing an error message sounds good, but I don't think it should `throw` an error here, for the same reason as above.

File: test/library/packages/NetCDF/pres_temp_4D_rd.chpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (lydia-duncan) [reply test/library/packages/NetCDF/pres_temp_4D_rd.chpl]: Okay, that seems reasonable

File: test/library/packages/NetCDF/pres_temp_4D_rd.chpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (lydia-duncan) [reply test/library/packages/NetCDF/pres_temp_4D_rd.chpl]: Okay, thanks!

**Comment by daviditen** [comment]
I just pushed a commit that prints error messages before each error return and moves those array declarations to their own lines.  Let me know if you see anything else you think needs improvement.

File: test/library/packages/NetCDF/pres_temp_4D_rd.chpl [hunk 3]
```diff
@@ -65,11 +66,15 @@ proc main {
 
    /* Check the coordinate variable data. */
    for lat in 0..#nlat do
-      if (lats[lat] != startLat + 5.0*lat) then
+      if lats[lat] != startLat + 5.0*lat {
+         writeln("Found a mismatched lattitued at #", lat);
```
Comment (lydia-duncan) [inline test/library/packages/NetCDF/pres_temp_4D_rd.chpl]: That indentation got real funky

Comment (lydia-duncan) [reply test/library/packages/NetCDF/pres_temp_4D_rd.chpl]: In more than just this location

**Comment by daviditen** [comment]
Stupid tabs.  I'll fix the indentation (and remove the tabs) with one more commit then go ahead and merge.  Thanks!

**Pull request merged**